tests/resolvers/Mutation/updateAdvertisement.spec.ts

```diff
@@ -0,0 +1,159 @@
+import type mongoose from "mongoose";
+import { Types } from "mongoose";
+import { Advertisement, User } from "../../../src/models";
+import type { MutationUpdateAdvertisementArgs } from "../../../src/types/generatedGraphQLTypes";
+import { connect, disconnect } from "../../helpers/db";
+import { updateAdvertisement as updateAdvertisementResolver } from "../../../src/resolvers/Mutation/updateAdvertisement";
+import {
+  ADVERTISEMENT_NOT_FOUND_ERROR,
+  USER_NOT_FOUND_ERROR,
+} from "../../../src/constants";
+import { beforeAll, afterAll, describe, it, expect, vi } from "vitest";
+import { createTestUser, type TestUserType } from "../../helpers/userAndOrg";
+
+type AdvertisementTypes = {
+  type: "POPUP" | "MENU" | "BANNER";
+  // Other properties specific to each type
+};
+
+// Define TestAdvertisementType
+type TestAdvertisementType = {
+  _id: string;
+  name: string;
+  link: string;
+  type: AdvertisementTypes;
+  startDate: string;
+  endDate: string;
+  // Add other fields based on your Advertisement model
+};
+
+// Function to create test advertisement
+const createTestAdvertisement = async (): Promise<TestAdvertisementType> => {
+  const testAdvertisementData = {
+    name: "Test Advertisement",
+    link: "https://example.com",
+    type: "POPUP",
+    startDate: "2023-01-01",
+    endDate: "2023-01-31",
+    // Add other fields based on your Advertisement model
+  };
+
+  // Create test advertisement in the database (assuming Advertisement model has create method)
+  const createdAdvertisement = await Advertisement.create(
+    testAdvertisementData
+  );
+
+  return {
```

Comment:
u can simply return `createdAdvertisement.object()`, also this dummy object generation should probaby be moved here https://github.com/PalisadoesFoundation/talawa-api/blob/main/tests/helpers

look at the files in that folder for examples

Reply:
made a new file named advertisement in the helpers' folder for dummy object creation and used proper variables in the updated codebase.